terminus-ui/scss/helpers/_color.spec.scss

```diff
@@ -9,80 +9,65 @@
     @include assert-equal($test, $expect);
   }
 
-  /*   @include test('should return the correct theme (this seems to fail due to class being added by true)') {
-    @include assert{
+  @include test('should return the correct theme and color for background-color') {
+    @include assert {
       @include output {
         @include theme-color(background-color);
       }
       @include expect {
-        ::ng-deep .u-theme--primary .test-output {
-          background-color: #00538a;
-          color: #fafafa;
-        }
-
-        ::ng-deep .u-theme--primary .test-output[disabled] {
-          background-color: #cecdd1;
-          color: #999;
-        }
-
-        ::ng-deep .u-theme--accent .test-output {
-          background-color: #2f854d;
-          color: #fafafa;
-        }
-
-        ::ng-deep .u-theme--accent .test-output[disabled] {
-          background-color: #cecdd1;
-          color: #999;
-        }
-
-        ::ng-deep .u-theme--error .test-output {
-          background-color: #c8604d;
-          color: #fafafa;
-        }
-
-        ::ng-deep .u-theme--error .test-output[disabled] {
-          background-color: #cecdd1;
-          color: #999;
-        }
-
-        ::ng-deep .u-theme--warn .test-output {
-          background-color: #dea83b;
-          color: #fafafa;
-        }
-
-        ::ng-deep .u-theme--warn .test-output[disabled] {
-          background-color: #cecdd1;
-          color: #999;
-        }
-
-        ::ng-deep .u-theme--highlight .test-output {
-          background-color: #efe557;
-          color: #302d35;
-        }
-
-        ::ng-deep .u-theme--highlight .test-output[disabled] {
-          background-color: #cecdd1;
-          color: #999;
-        }
-
-        ::ng-deep .u-theme--pure .test-output {
-          background-color: #fafafa;
+        @each $key, $value in $theme-and-color {
+          ::ng-deep .u-theme--#{$key} & {
+            background-color: $value;
+
+            &[disabled] {
+              background-color: color(utility, light);
+              color: color(utility);
+            }
+
+            @if (not(($key == 'pure')) and not(($key == 'highlight'))) {
+              color: $color__pure--light;
+            }
+
+            @if ($key == 'highlight') {
+              color: $color__pure--dark;
+            }
+          }
         }
+      }
+    }
+  }
 
-        ::ng-deep .u-theme--pure .test-output[disabled] {
-          background-color: #cecdd1;
-          color: #999;
+  @include test('should return the correct theme and color for color') {
+    @include assert {
+      @include output {
+        @include theme-color(color);
+      }
+      @include expect {
+        @each $key, $value in $theme-and-color {
+          ::ng-deep .u-theme--#{$key} & {
+            color: $value;
+
+            &[disabled] {
+              background-color: color(utility, light);
+              color: color(utility);
```

Comment:
Does this only test for `disabled` color, or this is like a skeleton for others to come along?

Reply:
It's a test of what we have now. The `theme-color` mixin sets the disabled pseudo class. 

Here's the code: https://github.com/GetTerminus/terminus-ui/blob/82e377d588a4e28aebe5cfe3f4bc2dddde1f5db4/terminus-ui/scss/helpers/_color.scss#L354